packages/extras/src/lib/components/controls/TrackballControls/TrackballControls.svelte

```diff
@@ -0,0 +1,54 @@
+<script lang="ts">
+  import { forwardEventHandlers, T, useTask, useParent, useThrelte } from '@threlte/core'
+  import type { Camera } from 'three'
+  import { TrackballControls as ThreeTrackballControls } from 'three/examples/jsm/controls/TrackballControls'
+  import { useControlsContext } from '../useControlsContext'
+  import type {
+    TrackballControlsEvents,
+    TrackballControlsProps,
+    TrackballControlsSlots
+  } from './TrackballControls.svelte'
+
+  type $$Props = TrackballControlsProps
+  type $$Events = TrackballControlsEvents
+  type $$Slots = TrackballControlsSlots
+
+  const parent = useParent()
+
+  const isCamera = (p: any): p is Camera => {
+    return p.isCamera
+  }
+
+  const { renderer, invalidate } = useThrelte()
+
+  if (!isCamera($parent)) {
+    throw new Error('Parent missing: <TrackballControls> need to be a child of a <Camera>')
+  }
+
+  export const ref = new ThreeTrackballControls($parent, renderer.domElement)
+
+  const { start, stop } = useTask(() => ref.update(), {
+    autoStart: true
+  })
+
+  const component = forwardEventHandlers()
+
+  const { trackballControls } = useControlsContext()
+</script>
+
+<T
+  is={ref}
+  let:ref
+  {...$$restProps}
+  bind:this={$component}
+  on:change={invalidate}
+  on:create={({ ref, cleanup }) => {
+    trackballControls.set(ref)
+    ref.update()
+    cleanup(() => {
+      trackballControls.set(undefined)
+    })
+  }}
```

Comment:
Since we're instantiating `ThreeTrackballControls` in the `<script>` we could move all of this there to more immediately set these values:

```svelte
<script>
  import { onDestroy } from 'svelte'
   ...
   const { trackballControls } = useControlsContext()
   export const ref = new ThreeTrackballControls($parent, renderer.domElement)
   trackballControls.set(ref)
   onDestroy(() => trackballControls.set(undefined))
   ...
```
And it should be fine to just call `ref.update()` in `useTask` since that is called before the first frame is rendered.

Reply:
cool, done!